Services/DataX.Config/DataX.Config/PublicService/JobOperation.cs

```diff
@@ -49,6 +49,19 @@ public async Task<Result> RestartJob(string jobName)
             return await this.SparkJobOperation.RestartJobWithRetries(jobName, timeout, _DefaultRetryInterval);
         }
 
+        public async Task<Result[]> RestartAllJobs()
+        {
+            var jobs = await SparkJobData.GetAll();
+            var results = new List<Result>();
+            foreach (var job in jobs)
+            {
+                var result = await RestartJob(job.Name);
```

Comment:
We should add a check to verify if the job was restarted fine. If we restart a lot of jobs in a small time interval we have seen failures in job restart where HDInsight fails to allocate resources to the jobs

Reply:
We should also add a retry logic in case of failures

Reply:
As we restart a batch of jobs in a given list of jobs, we can call SyncJobStateByNames() to check the status of each job in the batch and once all the jobs are in running state then proceed to next batch in the list. If some job was not in running state after a threshoold time period then we can issue a restart again for that job

Reply:
1. added a new API to wait for jobs up
2. added retry logics in the wait API